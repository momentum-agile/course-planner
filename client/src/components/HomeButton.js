import React from "react";
import { Link } from "react-router-dom";
import { IconButton } from "@chakra-ui/core";
import { AiFillHome } from "react-icons/ai";

const HomeButton = () => {
    return (
        <Link to="/">
            <IconButton as={AiFillHome} variantColor="black" left="20px" top="20px" size="xs" _hover={{ transform: "scale(1.1, 1.1)" }} />
        </Link>
    );
};

export default HomeButton;
