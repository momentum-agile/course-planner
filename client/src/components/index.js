import AppConfiguration from "./AppConfiguration.js";
import OutlineButton from "./OutlineButton.js";
import ProgrammeCard from "./ProgrammeCard.js";
import SearchBar from "./SearchBar.js";
import ProgrammeRequirementsItem from "./ProgrammeRequirementsItem.js";
import Table from "./Table.js";
import InlineEdit from "./InlineEdit";
import ConfirmationDialog from "./ConfirmationDialog";
import SearchSelect from "./SearchSelect";
import SingleSelect from "./SingleSelect";
import MenuWrapper from "./MenuWrapper";
import OptionsMenu from "./OptionsMenu";
import ExportMenu from "./ExportMenu";
import SaveCancelButtonSet from "./SaveCancelButtonSet";

export {
    AppConfiguration,
    OutlineButton,
    ProgrammeCard,
    SearchBar,
    ProgrammeRequirementsItem,
    Table,
    InlineEdit,
    SearchSelect,
    SingleSelect,
    ConfirmationDialog,
    OptionsMenu,
    MenuWrapper,
    ExportMenu,
    SaveCancelButtonSet,
};
