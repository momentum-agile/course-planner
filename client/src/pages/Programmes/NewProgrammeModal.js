import React, { useState, useEffect } from "react";
import {
    Input,
    AlertDialog,
    AlertDialogOverlay,
    AlertDialogContent,
    AlertDialogHeader,
    AlertDialogBody,
    AlertDialogFooter,
    useToast,
    Button,
} from "@chakra-ui/core";
import useProgrammmes from "./useProgrammes";
import { useHistory } from "react-router-dom";
import { colors as c } from "../../colors";

const NewProgrammeModal = ({ notifyAddition, isOpen, onClose }) => {
    const history = useHistory();
    const { createProgramme } = useProgrammmes();
    const [programmeName, setProgrammeName] = useState("");
    const [newProgramme, setNewProgramme] = useState({
        name: "",
        regulations: [],
        defaultPlan: null,
    });

    const cancelRef = React.useRef();
    const toast = useToast();

    useEffect(() => {
        setNewProgramme({
            name: programmeName,
            regulations: [],
            defaultPlan: null,
        });
    }, [programmeName]);

    const saveProgramme = async () => {
        await createProgramme(newProgramme, notifyAddition).then((res) => history.push(`/programmes/${res._id}`));
        onClose();

        toast({
            title: `Programme Created `,
            description: `Programme ${programmeName} has been successfully created ðŸŽ‰`,
            status: "success",
            duration: 5000,
            isClosable: true,
        });
        setProgrammeName("");
    };

    return (
        <>
            <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
                <AlertDialogOverlay />
                <AlertDialogContent>
                    <AlertDialogHeader fontSize="2xl" fontWeight="bold" width="100%">
                        Create New Programme
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        <Input
                            variant="filled"
                            placeholder="Enter New Programme Name"
                            textAlign="left"
                            value={programmeName}
                            onChange={(event) => setProgrammeName(event.target.value)}
                        />
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        <Button ref={cancelRef} onClick={onClose} color={c.darkGrey}>
                            Cancel
                        </Button>
                        <Button variantColor="blue" isDisabled={programmeName === ""} onClick={saveProgramme} ml={3}>
                            Create
                        </Button>
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialog>
        </>
    );
};

export default NewProgrammeModal;
