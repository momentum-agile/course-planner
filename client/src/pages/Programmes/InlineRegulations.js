import React, { useState, useEffect } from "react";
import {
    Flex,
    Text,
    Select,
    NumberInput,
    NumberInputField,
    NumberInputStepper,
    NumberIncrementStepper,
    NumberDecrementStepper,
    PseudoBox,
    IconButton,
    Tag,
    TagLabel,
    TagCloseButton,
} from "@chakra-ui/core";
import { AiOutlineCheckCircle, AiOutlineCloseCircle } from "react-icons/ai";
import useCourses from "../Courses/useCourses";
import useProgrammes from "./useProgrammes";
import { SingleSelect } from "../../components";
import { colors as c } from "../../colors";
import { useHorizontalScroll } from "./scrollHook";

const InlineRegulations = ({ closeEdit, programme, notifyUpdate, regulation }) => {
    const { data } = useCourses();
    const { updateProgramme } = useProgrammes();
    const [courses, setCourses] = useState([]);
    const [points, setPoints] = useState(15);
    const [pointRequirement, setPointRequirement] = useState("EXACTLY");

    const handleRemoveCourse = (courseId) => {
        const filteredCourses = courses.filter((course) => course._id !== courseId);
        setCourses(filteredCourses);
    };

    const handleUpdateRegulation = () => {
        if (courses.length !== 0 && points > 0) {
            if (regulation === undefined) {
                programme.regulations.push({ courses, points, pointRequirement });
            } else {
                const foundRegulationIndex = programme.regulations.findIndex((r) => r._id === regulation._id);
                programme.regulations[foundRegulationIndex] = {
                    ...programme.regulations[foundRegulationIndex],
                    courses,
                    points,
                    pointRequirement,
                };
            }
            updateProgramme(programme, notifyUpdate);
            closeEdit();
        }
    };

    useEffect(() => {
        if (regulation !== undefined && regulation.courses) {
            setCourses(data.filter((course) => regulation.courses.includes(course._id)));
            setPointRequirement(regulation.pointRequirement);
            setPoints(regulation.points);
        } else {
            setCourses([]);
            setPointRequirement("EXACTLY");
            setPoints(15);
        }
    }, [regulation, data]);

    const scrollRef = useHorizontalScroll();

    return (
        <PseudoBox
            height="50px"
            width="100%"
            paddingLeft="3px"
            paddingRight="3px"
            bg={c.white}
            borderRadius="2px"
            borderBottom={`solid 1px ${c.whiteGrey}`}
            flex-direction="row"
            alignItems="center"
            textAlign="center"
            justify="space-around"
            color={c.midnightBlue}
        >
            <Flex justify="space-around" align="center" height="100%">
                {/* Dropdown for Requirement Clause */}
                <Select
                    width="17%"
                    height="30px"
                    bg={c.white}
                    value={pointRequirement}
                    cursor="pointer"
                    onChange={(e) => setPointRequirement(e.target.value)}
                >
                    <option value="EXACTLY">Exactly</option>
                    <option value="ATLEAST">At Least</option>
                    <option value="UPTO">Up to</option>
                </Select>

                {/* Number input for Points */}
                <NumberInput
                    step={5}
                    min={0}
                    height="40px"
                    width="13%"
                    bg={c.white}
                    ml={-3}
                    value={points}
                    onChange={(value) => setPoints(value)}
                >
                    <NumberInputField bg={c.white} />
                    <NumberInputStepper>
                        <NumberIncrementStepper />
                        <NumberDecrementStepper />
                    </NumberInputStepper>
                </NumberInput>

                <Text textAlign="left" width="3%" fontSize="10px" color={c.darkBlue} ml={-3}>
                    FROM
                </Text>

                {/* Course Dropdown */}
                <Flex direction="row" width="55%" align="center">
                    <SingleSelect
                        onChange={(course) => setCourses([...courses, course])}
                        data={data
                            .filter((course) => !courses.includes(course))
                            .map((course) => ({ value: course, label: course.courseCode }))}
                        placeholder="Search"
                    />
                    <Flex
                        className="programmeRequirements"
                        minHeight="40px"
                        maxHeight="70px"
                        width="100%"
                        overflowX="scroll"
                        whiteSpace="nowrap"
                        padding="5px"
                        direction="row"
                        ml={2}
                        ref={scrollRef}
                    >
                        {courses.reverse().map((course) => (
                            <Flex>
                                <Tag size="sm" key={course._id} rounded="full" variant="solid" variantColor="cyan" mr={1}>
                                    <TagLabel>{course.courseCode}</TagLabel>
                                    <TagCloseButton onClick={() => handleRemoveCourse(course._id)} />
                                </Tag>
                            </Flex>
                        ))}
                    </Flex>

                    <Flex direction="row" justify="flex-end" align="flex-end" ml={2}>
                        <IconButton
                            as={AiOutlineCloseCircle}
                            size="xs"
                            color="red.400"
                            onClick={closeEdit}
                            cursor="pointer"
                            mr={1}
                            bg="none"
                        />
                        <IconButton
                            as={AiOutlineCheckCircle}
                            size="xs"
                            isDisabled={!(courses.length !== 0 && points > 0)}
                            color={c.green}
                            onClick={() => handleUpdateRegulation()}
                            cursor="pointer"
                            bg="none"
                        />
                    </Flex>
                </Flex>
            </Flex>
        </PseudoBox>
    );
};

export default InlineRegulations;
