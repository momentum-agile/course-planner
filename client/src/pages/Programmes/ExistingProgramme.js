import React, { useEffect, useState } from "react";
import { Box, Button, Flex, Icon, IconButton, Input, Stack, Text } from "@chakra-ui/core";
import { ConfirmationDialog, MenuWrapper, OutlineButton, ProgrammeRequirementsItem, SaveCancelButtonSet } from "../../components";
import { useHistory } from "react-router-dom";
import useProgrammmes from "./useProgrammes";
import useCourses from "../Courses/useCourses";
import { colors as c } from "../../colors";
import InlineRegulations from "./InlineRegulations";

const ExistingProgramme = ({ programme, notifyUpdate }) => {
    const history = useHistory();
    const { deleteProgramme, updateProgramme, createProgrammeDegreePlan } = useProgrammmes();
    const { data } = useCourses();
    const [programmeDegreeInfo, setProgrammeDegreeInfo] = useState({});
    const [createNewRequirements, setCreateNewRequirements] = useState(false);
    const [editRequirements, setEditRequirements] = useState(-1);
    const [openConfirmationDialog, setOpenConfirmationDialog] = useState(false);
    const [openConfirmationDeleteRegulation, setOpenConfirmationDeleteRegulation] = useState(false);
    const [regulationtoDelete, setRegulationToDelete] = useState({});
    const [regulationInEdit, setRegulationInEdit] = useState(undefined);
    const [isEdited, setIsEdited] = useState(false);
    const [name, setName] = useState("");

    useEffect(() => {
        setProgrammeDegreeInfo(programme);
        setName(programme.name);
    }, [isEdited, programme]);

    const cancelEditProgramme = () => {
        setIsEdited(false);
        setName(programmeDegreeInfo.name);
    };

    const saveEditProgramme = () => {
        const editedProgramme = {
            ...programmeDegreeInfo,
            name: name,
        };

        // TODO: show popup success message
        setIsEdited(false);

        // Update the programme and refresh
        updateProgramme(editedProgramme, notifyUpdate);
    };

    const confirmDelete = () => {
        setOpenConfirmationDialog(false);
        deleteProgramme(programmeDegreeInfo._id, history, notifyUpdate);
    };

    const deleteRegulation = (regulation) => {
        setRegulationToDelete(regulation);
        setOpenConfirmationDeleteRegulation(true);
    };

    const confirmDeleteRegulation = (regulation) => {
        const updatedRegulations = programme.regulations.filter((r) => r._id !== regulation._id);
        const updatedProgramme = { ...programme, regulations: updatedRegulations };
        updateProgramme(updatedProgramme, notifyUpdate);
        setOpenConfirmationDeleteRegulation(false);
        setCreateNewRequirements(false);
        setEditRequirements(-1);
    };

    return (
        <Flex width="100%" direction="column">
            <Flex align="center" justify="center" direction="row">
                <Flex align="center" justify="center" marginTop="10px">
                    {isEdited ? (
                        <Input
                            variant="flushed"
                            placeholder="Programme Name"
                            textAlign="center"
                            width="100%"
                            size="lg"
                            value={name}
                            onChange={(event) => setName(event.target.value)}
                        />
                    ) : (
                        <Text textAlign="center" fontSize="5xl" color={c.nightBlue} as="b">
                            {programmeDegreeInfo.name}
                        </Text>
                    )}
                </Flex>

                <Flex justify="flex-end" align="flex-end" right="50px" position="absolute">
                    {/* Dropdown menu to edit and delete */}
                    <MenuWrapper
                        item={programmeDegreeInfo}
                        itemType="Programme"
                        setOpenConfirmationDialog={setOpenConfirmationDialog}
                        openConfirmationDialog={openConfirmationDialog}
                        confirm={confirmDelete}
                        onEdit={!isEdited ? () => setIsEdited(true) : undefined}
                    />
                </Flex>
            </Flex>

            {/* Save and Close Buttons for Editing */}
            {isEdited ? (
                <Flex justifyContent="space-evenly" width="100%" align="center" mt={8} mb={3}>
                    <SaveCancelButtonSet isActive={name !== ""} onCancel={cancelEditProgramme} onSave={saveEditProgramme} />
                </Flex>
            ) : (
                ""
            )}

            {/* PROGRAMME REGULATIONS */}
            <Flex align="center" justify="center" direction="column">
                <Text textAlign="center" fontSize="md" color={c.nightBlue} mt={5}>
                    Regulations
                </Text>
                <Box bg={c.whiteGrey} width="75%" textAlign="center" padding="5px" marginTop="10px" className="regulationsBox">
                    {/* Add Regulations Button to activate Component via createNewRequirements */}
                    {!createNewRequirements ? (
                        <IconButton
                            mt={2}
                            mb={2}
                            width="100%"
                            icon="add"
                            bg={c.midnightBlue}
                            color={c.white}
                            _hover={{
                                bg: c.nightBlue,
                            }}
                            onClick={() => {
                                setCreateNewRequirements(true);
                                setRegulationInEdit(undefined);
                            }}
                        />
                    ) : (
                        <InlineRegulations
                            closeEdit={() => setCreateNewRequirements(false)}
                            programme={programme}
                            notifyUpdate={notifyUpdate}
                        />
                    )}
                    <Stack direction="column" height="500px" width="100%" overflowY="auto" align="center">
                        {programme.regulations.map((regulation, index) =>
                            editRequirements !== index ? (
                                <ProgrammeRequirementsItem
                                    key={index}
                                    itemNumber={`${index + 1}.`}
                                    onEdit={() => {
                                        setEditRequirements(index);
                                        setRegulationInEdit(regulation);
                                    }}
                                    pointRequirement={regulation.pointRequirement}
                                    points={regulation.points}
                                    courseList={data.filter((course) => regulation.courses.includes(course._id))}
                                    deleteButton={
                                        <Button
                                            _hover={{ transform: "scale(1.2, 1.2)" }}
                                            _active="none"
                                            bg="none"
                                            align="flex-reverse"
                                            justify="flex-end"
                                            onClick={() => deleteRegulation(regulation)}
                                        >
                                            <Icon name="small-close" />
                                        </Button>
                                    }
                                />
                            ) : (
                                <InlineRegulations
                                    closeEdit={() => setEditRequirements(-1)}
                                    programme={programme}
                                    notifyUpdate={notifyUpdate}
                                    regulation={regulationInEdit}
                                />
                            ),
                        )}
                    </Stack>
                    <ConfirmationDialog
                        item={regulationtoDelete}
                        itemType={"Regulation"}
                        action={"Delete"}
                        isOpen={openConfirmationDeleteRegulation}
                        onClose={() => setOpenConfirmationDeleteRegulation(false)}
                        confirm={confirmDeleteRegulation}
                    />
                </Box>
            </Flex>

            <Flex align="center" justify="center" width="100%">
                {programmeDegreeInfo.defaultPlan ? (
                    <OutlineButton text="View Template" to={`/plan/${programmeDegreeInfo.defaultPlan}`} height="60px" />
                ) : (
                    <OutlineButton
                        text="Define Template"
                        onClick={() => createProgrammeDegreePlan(programme).then((res) => history.push(`/plan/${res._id}`))}
                        height="60px"
                    />
                )}
            </Flex>
        </Flex>
    );
};

export default ExistingProgramme;
