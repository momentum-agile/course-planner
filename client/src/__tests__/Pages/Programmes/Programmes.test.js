import React from "react";
import renderer from "react-test-renderer";
import { AppConfiguration } from "../../../components";
import { Programmes } from "../../../pages";
import NewProgrammeModal from "../../../pages/Programmes/NewProgrammeModal";
import ExistingProgramme from "../../../pages/Programmes/ExistingProgramme";
import EmptyProgramme from "../../../pages/Programmes/EmptyProgramme";
import InlineRegulations from "../../../pages/Programmes/InlineRegulations";

test("Main Programmes Page", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <Programmes />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("Creating programmes should be hidden unless acted on", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <NewProgrammeModal />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("Existing programmes sub-page ", () => {
    let programme = {
        name: "SOFTENG761",
        regulations: [
            {
                points: 15,
                pointRequirement: "ATLEAST",
                courses: [],
            },
        ],
        defaultPlan: null,
    };

    const tree = renderer
        .create(
            <AppConfiguration>
                <ExistingProgramme id={"fakeProgrammeID"} programme={programme} />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("Empty programmes sub-page ", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <EmptyProgramme />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("Inline Regulations ", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <InlineRegulations />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
