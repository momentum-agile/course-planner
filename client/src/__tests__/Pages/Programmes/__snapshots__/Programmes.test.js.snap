// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating programmes should be hidden unless acted on 1`] = `null`;

exports[`Empty programmes sub-page  1`] = `
<div
  className="css-7eifx2"
>
  <div
    className="css-gmuwbf"
  >
    <i
      className="css-1fwnx19"
    >
      Create or Select a Programme to view its information here.
    </i>
  </div>
</div>
`;

exports[`Existing programmes sub-page  1`] = `
<div
  className="css-1ofqig9"
>
  <div
    className="css-1hyoz7m"
  >
    <div
      className="css-1yfye7s"
    >
      <b
        className="css-1eiq96g"
      >
        SOFTENG761
      </b>
    </div>
    <div
      className="css-qzt8sy"
    >
      <div
        className="css-k008qs"
      >
        <div
          className="css-k008qs"
        >
          <button
            aria-controls="menu-4"
            aria-expanded={false}
            aria-haspopup="menu"
            className="css-992tcj"
            id="menubutton-5"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <svg
              className="css-11orafl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              type="button"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S13.1 10 12 10zM18 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S19.1 10 18 10zM6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S7.1 10 6 10z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  
  <div
    className="css-u6v8er"
  >
    <p
      className="css-15uw6ol"
    >
      Regulations
    </p>
    <div
      className="regulationsBox css-1obiqbc"
    >
      <button
        className="css-lcsfhn"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="css-1im46kq"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        className="css-dscpp4"
      >
        <div
          className="css-vipb1e"
          onClick={[Function]}
        >
          <div
            className="css-tkevr6"
          >
            <b
              className="css-1eqa46m"
            >
              1.
            </b>
            <p
              className="css-zs3z9w"
            >
              At least
            </p>
            <p
              className="css-zs3z9w"
            >
              15 pts
            </p>
            <p
              className="css-1vkqwb0"
            >
              FROM
            </p>
            <div
              className="programmeRequirements css-1jknthz"
            >
              <div
                className="css-nl7a3z"
              />
            </div>
            <div
              className="css-1i00gen"
            >
              <button
                className="css-1h1blff"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-1im46kq"
                  focusable={false}
                  role="presentation"
                  viewBox="0 0 16 16"
                >
                  <path
                    d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 0 0-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L9.41 8z"
                    fill="currentColor"
                    fillRule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1lads1q"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="css-rshclt"
        onClick={[Function]}
        type="button"
      >
        <div
          className="css-gmuwbf"
        >
          <p
            className="css-fdblai"
          >
            Define Template
          </p>
        </div>
      </button>
    </a>
  </div>
</div>
`;

exports[`Inline Regulations  1`] = `
<div
  className="css-1vqyz7j"
>
  <div
    className="css-1lkkk2k"
  >
    <div
      className="css-1myz3ur"
    >
      <select
        className="css-db3xwu"
        onChange={[Function]}
        value="EXACTLY"
      >
        <option
          value="EXACTLY"
        >
          Exactly
        </option>
        <option
          value="ATLEAST"
        >
          At Least
        </option>
        <option
          value="UPTO"
        >
          Up to
        </option>
      </select>
      <div
        className="css-1lepn2w"
      >
        <svg
          aria-hidden="true"
          className="css-12c4avn"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      className="css-abll83"
    >
      <input
        aria-invalid={false}
        aria-valuemax={Infinity}
        aria-valuemin={0}
        aria-valuenow={15}
        autoComplete="off"
        className="css-8r3r98"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="spinbutton"
        type="text"
        value={15}
      />
      <div
        aria-hidden={true}
        className="css-1fjocbx"
      >
        <div
          className="css-j3jbfm"
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          role="button"
        >
          <svg
            className="css-1v61v6x"
            focusable={false}
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          className="css-j3jbfm"
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          role="button"
        >
          <svg
            className="css-1v61v6x"
            focusable={false}
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <p
      className="css-1bpicay"
    >
      FROM
    </p>
    <div
      className="css-14w3xfh"
    >
      <div
        className=" css-2b097c-container"
        onKeyDown={[Function]}
      >
        <div
          className=" css-16uwxe8-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-g1d714-ValueContainer"
          >
            <div
              className=" css-1wa3eu0-placeholder"
            >
              Search
            </div>
            <div
              className="css-rga2ib-Input"
            >
              <div
                className=""
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  disabled={false}
                  id="react-select-2-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "boxSizing": "content-box",
                      "color": "inherit",
                      "fontSize": "inherit",
                      "label": "input",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "1px",
                    }
                  }
                  tabIndex="0"
                  type="text"
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-6q0nyr-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="programmeRequirements css-133uvjq"
      />
      <div
        className="css-18y9g4n"
      >
        <svg
          className="css-9js0pz"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          type="button"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 0 0-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
          />
          <path
            d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
        </svg>
        <svg
          aria-disabled={true}
          className="css-1c0ztg6"
          disabled={true}
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          type="button"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
          />
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main Programmes Page 1`] = `
<div
  className="css-qxc4md"
>
  <div
    className="css-1lt3qay"
  >
    <div
      className="css-1u9fuhx"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <svg
          className="css-1hd2lmz"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          type="button"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
          />
        </svg>
      </a>
    </div>
    <div
      className="css-9jtr4b"
    >
      <p
        className="css-wqr37p"
      >
        Programmes
      </p>
    </div>
    <div
      className="css-9jn3o8"
    >
      <div
        className="css-1brfyz3"
      >
        <div
          className="css-1lads1q"
        >
          <div
            className="css-aetr73"
          >
            <div
              className="css-k6882c"
            >
              <svg
                className="css-1xb1rov"
                focusable={false}
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              className="css-n5gi9c"
              onChange={[Function]}
              placeholder=""
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="programmesBox css-1yy8858"
    >
      <div
        className="css-1deo6ki"
      >
        <button
          className="css-16a58ri"
          onClick={[Function]}
          type="button"
        >
          <p
            className="css-1li4w8y"
          >
            <svg
              className="css-p9gibj"
              focusable={false}
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </p>
        </button>
      </div>
    </div>
  </div>
  <div
    className="css-6hu3eg"
  >
    <div
      className="css-trwzgo"
    >
      <div
        className="css-7eifx2"
      >
        <div
          className="css-gmuwbf"
        >
          <i
            className="css-1fwnx19"
          >
            Create or Select a Programme to view its information here.
          </i>
        </div>
      </div>
    </div>
  </div>
</div>
`;
